File: SuperNumber/Java/SuperNumber.java
public class SuperNumber
{
    public static void main(String[] args)
    {
        long startTime = System.currentTimeMillis();
        int[] superNumbers = new int[9];
        for (int j = 2; j <= 9; j++)
        {
            superNumbers = new int[j];
            int possibilityCount = (int)Math.pow(j, j);
            for (int i = 0; i < possibilityCount - 1; i++)
            {
                incrementArray(superNumbers, superNumbers.length - 1);
                if (isSuperNumber(superNumbers))
                {
                    for(int digit : superNumbers)
                        System.out.print(digit);
                    System.out.println();
                }
            }
        }
        long stopTime = System.currentTimeMillis();
        long elapsedTime = (stopTime - startTime)/1000;
        System.out.print("\nTime Taken(Seconds):"+elapsedTime);
    }
    public static void incrementArray(int[] possibilities, int digitIndex)
    {
        int tmp = 0;
        int digitCount = possibilities.length;
        tmp = possibilities[digitIndex] + 1;
        if (tmp % digitCount == 0)
        {
            possibilities[digitIndex] = 0;
            incrementArray(possibilities, digitIndex - 1);
        }
        else
            possibilities[digitIndex] = tmp;
    }
    public static boolean isSuperNumber(int[] superNumbers)
    {
        int count = 0;
        int digitCount = superNumbers.length;
        for (int index = 0; index < digitCount; index++)
        {
            count = 0;
            for (int digit : superNumbers)
            {
                if (digit == index)
                    count++;
            }
            if (count != superNumbers[index])
                return false;
        }
        return true;
    }
}

File: SuperNumber/Scala/SuperNumber.scala

object SuperNumber
{
	def main(args: Array[String])
	{
		val start=System.nanoTime()
		
		for(j<-2 to 9)
		{
			var superNumbers=new Array[Int](j)
			val possibilityCount : Int=math.pow(j,j).toInt
			for(i<-0 until possibilityCount-1)
			{
				incrementArray(superNumbers,superNumbers.length-1)
				if(isSuperNumber(superNumbers))
				{
					superNumbers.foreach(print)
					println
				}
			}
		}
		val endTime :Double=System.nanoTime().toDouble
		val seconds :Double = (endTime-start) / 1000000000.0d

		println("Total Seconds:"+seconds)
	}

	def incrementArray(possibilities: Array[Int],digitIndex:Int)
	{
		var tmp:Int=0
		var digitCount:Int = possibilities.length
		tmp=possibilities(digitIndex)+1
			if(tmp%digitCount==0)
			{
				possibilities(digitIndex)=0
				incrementArray(possibilities,digitIndex-1)
			}
			else
				possibilities(digitIndex)=tmp
	}
	def isSuperNumber(superNumbers :Array[Int]) : Boolean  =	{
		var count:Int=0
		val digitCount=superNumbers.length
		for(index<-0 until digitCount)
		{
			count=0
			superNumbers.foreach((digit)=>{
				if (digit == index)
					count=count+1
			})
			
			if(count!=superNumbers(index))
				return false
		}
		return true
	}
	
}